contrib_bots/contrib_server.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+from flask import Flask, request
+from .run import get_lib_module
+from .bot_lib import BotHandlerApi
+import os
+import sys
+import json
+from typing import Any, Dict, Mapping
+
+our_dir = os.path.dirname(os.path.abspath(__file__))
+
+# For dev setups, we can find the API in the repo itself.
+if os.path.exists(os.path.join(our_dir, '../api/zulip')):
+    sys.path.insert(0, '../api')
+
+from zulip import Client
+
+bots_config = {
+    # the bot configuration will be added in this way
+    # "followup": {
+    #     "email": "followup-bot@zulip.localhost",
+    #     "key": "j5izMcpnW3SfkMlovez4Ihv8Usxrk9NO",
+    #     "site": "http://localhost:9991"
+    # }
+} # type: Dict[str, Mapping[str, str]]
+available_bots = bots_config.keys()
+bots_lib_module = {} # type: Dict[str, Any]
+
+for bot in available_bots:
+    path = "bots/" + str(bot) + "/" + str(bot) + ".py"
+    bots_lib_module[bot] = get_lib_module(path)
+
+class StateHandler(object):
+    def __init__(self):
+        # type: () -> None
+        self.state = None
+
+    def set_state(self, state):
+        # type: (Any) -> None
+        self.state = state
+
+    def get_state(self):
+        # type: () -> Any
+        return self.state
+
+app = Flask(__name__)
+
+@app.route('/')
+def main():
+    # type: () -> str
+    return ""
+
+@app.route('/bots/<bot>', methods=['POST'])
+def test(bot):
+    # type: (str) -> str
+    # if bot not in available_bots:
+    #     return "requested bot service not supported"
+
+    client = Client(email=bots_config[bot]["email"],
+                    api_key=bots_config[bot]["key"],
+                    site=bots_config[bot]["site"])
+    restricted_client = BotHandlerApi(client)
+    message_handler = bots_lib_module[bot].handler_class()
+    state_handler = StateHandler()
+
+    event = json.loads(request.data)
+    message_handler.handle_message(message=event["message"],
```

Comment:
Yeah it would be helpful to add the comment. Currently we are sending the data directly but later on I think we would be passing it through an interface designed for this server. That might be the best place for it. Do you want me to add a TODO comment inside DoRestCall?